hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/snapshot/SnapshotManager.java

```diff
@@ -541,8 +541,8 @@ public void clearSnapshottableDirs() {
    *
    * @return maximum allowable snapshot ID.
    */
-   public int getMaxSnapshotID() {
-    return ((1 << SNAPSHOT_ID_BIT_WIDTH) - 1);
+  public int getMaxSnapshotID() {
+    return (1 << SNAPSHOT_ID_BIT_WIDTH);
```

Comment:
I think we can still leave this as max index -1, as we would only have one snapshot id here.